src/cogent3/app/io.py

```diff
@@ -139,42 +140,18 @@ def findall(base_path, suffix="fa", limit=None, verbose=False):
 def get_data_store(
     base_path: Union[str, pathlib.Path], suffix=None, limit=None, verbose=False
 ):
-    """returns DataStore containing glob matches to suffix in base_path
-
-    Parameters
-    ----------
-    base_path : str or Path
-        path to directory or zipped archive
-    suffix : str
-        suffix of filenames
-    limit : int or None
-        the number of matches to return
-    Returns
-    -------
-    ReadOnlyDirectoryDataStore or ReadOnlyZippedDataStore
-    """
-    base_path = pathlib.Path(base_path)
-    base_path = base_path.expanduser().absolute()
-    if base_path.suffix in (".tinydb", ".sqlitedb"):
-        suffix = "json"
-
-    if suffix is None:
-        raise ValueError("suffix required")
-
-    if not base_path.exists():
-        raise ValueError(f"'{base_path}' does not exist")
-
-    if type(suffix) != str:
-        raise ValueError(f"{suffix} is not a string")
-
-    if zipfile.is_zipfile(base_path):
-        ds_suffix = ".zip"
-    elif base_path.suffix:
-        ds_suffix = base_path.suffix
-    else:
-        ds_suffix = None
-    klass = _datastore_reader_map[ds_suffix]
-    return klass(base_path, suffix=suffix, limit=limit)
+    """DEPRECATED, use io_new.open_data_store"""
+    warnings.warn(
```

Comment:
please use the `cogent3` interface to warnings since I search for those before each release.

There are 2 convenience functions:

- `deprecated`
- `discontinued`

In this case, you want
```python
from cogent3.util.warning import deprecated
```
In this case, you would do
```python
deprecated("function", "get_data_store", "cogent3.open_data_store", "2023.3", "renamed and now a top-level import", stack_level=1)
```

The latter also means, of course, that in `cogent3.__init__` you should add the following statement
```python
from cogent3.app.io_new import open_data_store
```
Although there may be a circular import issue

Reply:
Note, I never put a day on the deprecation version. Month is sufficient.